x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/TrainedModelDeploymentTask.java

```diff
@@ -80,15 +82,11 @@ public TaskParams getParams() {
         return params;
     }
 
-    public void stop(String reason) {
-        logger.debug("[{}] Stopping due to reason [{}]", getModelId(), reason);
-        licensedFeature.stopTracking(licenseState, "model-" + params.getModelId());
```

Comment:
I think this still needs to be called. If you are concerned, maybe wrap the listener and call this on response/failure?

Reply:
Yeah I'm not happy with this pattern. The task asks the node service to stop then the node service calls back to `TrainedModelDeploymentTask::markAsStopped` from `stopDeploymentAsync`. This means `markAsStopped` is a public method which doesn't make much sense as part of the public API.

The problem is there are a few ways `TrainedModelAllocationNodeService::stopDeploymentAsync` can be called such as the service noticing the deployment has been deleted or the task being cancelled. 

I deleted these lines because  `markAsStopped` (formerly `stopWithoutNotification`) was being called anyway and the same work occurs there.

Reply:
> I deleted these lines because markAsStopped (formerly stopWithoutNotification) was being called anyway and the same work occurs there.

Gotcha, we just need to be careful and make sure that `stopTracking` is being called.